### ISSUE TITLE
addStylesToShadowDOM.js error for WebComponent target

### ISSUE THREAD

**Issue by matrunchyk** [vuejs/vue-style-loader#23]
Version: vue-style-loader@4.1.0

## Bug report

Steps to reproduce:
1. `vue create vue-style-loader-bug`
2. `cd vue create vue-style-loader-bug`
3. Rename `App.vue` -> `Example-App.vue` (since WC should have at least 2 words)
4. By default `Example-App.vue` contains one `<HelloWorld>` tag. Add one or few more. This is important. So the file should look like:

```
<div id="app">
    <img src="./assets/logo.png">
    <HelloWorld msg="Welcome to Your Vue.js App"/>
    <HelloWorld msg="Welcome to Your Vue.js App"/>
  </div>
```
5. Run `./node_modules/.bin/vue-cli-service build --target wc --name example-app src/Example-App.vue` to generate WebCompontent to `dist` folder.
6. Open `dist/demo.html` in Chrome. 
7. Open Chrome console tools. You'll see the bug:
![Alt text](https://monosnap.com/file/ZTsyEmED3sOVyFfuoCCH3jO5zwDnAg.png)

This happen because we expect `style.parts[j]` to be a function and [execute it](https://github.com/vuejs/vue-style-loader/blob/master/lib/addStylesShadow.js#L31) which is not in fact a function, but `undefined` ([a proof while debugging](https://monosnap.com/file/kFajbYmPmFZ3N5U9tV5GyNPDDJt4ea))

The reason why `style.parts[j]` is `undefined`, [because we run](https://github.com/vuejs/vue-style-loader/blob/master/lib/addStylesShadow.js#L42) `parts.push()` passing a result of `addStyle()` function which [DOES NOT return anything](https://github.com/vuejs/vue-style-loader/blob/master/lib/addStylesShadow.js#L80)!

Please note, that it won't fail if you don't use multiple vue components within the same file (they should have a `<style>` tag), because the `if` condition [will not run in that case](https://github.com/vuejs/vue-style-loader/blob/master/lib/addStylesShadow.js#L28).

I notices there are no tests for this file at all. 

@yyx990803, is there any possibility to fix it? Now it's impossible to use multiple components within 1 component for Web Components.

Thanks.

**Comment by jgribonvald** [comment]
Hi this bug is a problem for me too.

Do you have a way to avoid this problem when using severall components ?
Thanks

**Comment by altitudems** [comment]
I can confirm the same issue here, kind of a show stopper for using web components

**Comment by yyx990803** [comment]
Fixed in 4.1.1
